js/gateio.js

```diff
@@ -2256,6 +2256,127 @@ module.exports = class gateio extends Exchange {
         };
     }
 
+    parsePosition (position, market = undefined) {
+        //
+        //     {
+        //         value: "12.475572",
+        //         leverage: "0",
+        //         mode: "single",
+        //         realised_point: "0",
+        //         contract: "BTC_USDT",
+        //         entry_price: "62422.6",
+        //         mark_price: "62377.86",
+        //         history_point: "0",
+        //         realised_pnl: "-0.00624226",
+        //         close_order:  null,
+        //         size: "2",
+        //         cross_leverage_limit: "25",
+        //         pending_orders: "0",
+        //         adl_ranking: "5",
+        //         maintenance_rate: "0.005",
+        //         unrealised_pnl: "-0.008948",
+        //         user: "663337",
+        //         leverage_max: "100",
+        //         history_pnl: "14.98868396636",
+        //         risk_limit: "1000000",
+        //         margin: "0.740721495056",
+        //         last_close_pnl: "-0.041996015",
+        //         liq_price: "59058.58"
+        //     }
+        //
+        const contract = this.safeValue (position, 'contract');
+        market = this.safeMarket (contract, market);
+        const now = this.milliseconds ();
+        const size = this.safeValue (position, 'size');
+        const side = size > 0 ? 'buy' : 'sell';
+        const maintenance_rate = this.safeValue (position, 'maintenance_rate');
+        // const markPrice = this.safeValue (position, 'mark_price');
+        // const contractSize = this.safeValue (position, 'size');
+        const notional = this.safeValue (position, 'value');
+        return {
+            'info': position,
+            'symbol': market['symbol'],
+            'timestamp': now,
+            'datetime': this.iso8601 (now),
+            'initialMargin': this.safeValue (position, 'margin'),
+            'initialMarginPercentage': undefined,
+            'maintenanceMargin': maintenance_rate * notional,
+            'maintenanceMarginPercentage': maintenance_rate * 100,
+            'entryPrice': this.safeValue (position, 'entry_price'),
+            'notional': notional,
+            'leverage': this.safeValue (position, 'leverage'),
+            'unrealizedPnl': this.safeValue (position, 'unrealised_pnl'),
+            'contracts': undefined,
```

Comment:
i don't understand what this property is

Reply:
I think it's the size of a position in contracts.

Reply:
I may be wrong about it, looking forward to @frosty00 's comments on it.

Reply:
There's a `contractSize` property right below it though

Reply:
is the contractSize supposed to be how much each contract is worth? I don't believe that's given

Reply:
Ok, I can get the value of a contract, by dividing the total value by the size

Reply:
> Ok, I can get the value of a contract, by dividing the total value by the size

And then I have to divide this by the mark price to get the size in the base currency